### ISSUE TITLE
In dev branch the CLI doesn't exit when a message fails during execution

### ISSUE THREAD

**Issue by il3ven** [neume-network/strategies#184]
This is a bug in the new implementation of lifecycle done in neume-network/strategies#168. When a message from the extraction has the error property set the extract handler is not called. From the perspective of extract handler it is still waiting for the message and hence never resolves.

```js
worker.on("message", async (message) => {
  if (message.error) {
    throw new Error(message.commissioner + ":" + message.error);
  }

  // We never reach here
  // Even if we reach here message.commisioner is undefined
  messageRouter.emit(`${message.commissioner}-extraction`, message);
});
```

**Comment by TimDaub** [comment]
> [TimDaub](/TimDaub) closed this as [completed](/neume-network/strategies/issues?q=is%3Aissue+is%3Aclosed+archived%3Afalse+reason%3Acompleted) in [ #183 ](/neume-network/strategies/pull/183) [23 minutes ago](#event-7062771951)

I think this was closed incorrectly, wasn't it @il3ven?

**Comment by il3ven** [comment]
The bug was solved in https://github.com/neume-network/strategies/pull/183/commits/8288c2dd7fa76f1b4d6458cae0d8db3f557a02a4 and https://github.com/neume-network/extraction-worker/pull/37. The CLI is now able exit properly.

A related issue that still remains is how to handle error messages. In https://github.com/neume-network/strategies/pull/183/commits/8288c2dd7fa76f1b4d6458cae0d8db3f557a02a4 I am simply logging them.

**Comment by TimDaub** [comment]
> The bug was solved in [8288c2d](https://github.com/neume-network/strategies/commit/8288c2dd7fa76f1b4d6458cae0d8db3f557a02a4) and [neume-network/extraction-worker#37](https://github.com/neume-network/extraction-worker/pull/37). The CLI is now able exit properly.

I just released extraction-worker@0.5.0 and it should fix it.
